build.ps1

```diff
@@ -101,7 +107,18 @@ function AddDiaSymReaderToPath()
             $_ -replace '\s+Base Path:',''
         }
 
-    $diaSymPath = Join-Path $sdkBasePath.Trim() "Roslyn\bincore\runtimes\win\native"
+    $parent = Split-Path -Path $sdkBasePath.Trim()
+    $maxValue = 0
+    Get-ChildItem $parent\2.2.* | 
+        ForEach-Object {
+            $newVal = $_.Extension -replace '\.',''
+            if($newVal -gt $maxValue) {
+                $maxValue = $newVal
+            }
+        }
+        
+    $finalPath = $parent + "\2.2.$maxValue"
+    $diaSymPath = Join-Path $finalPath.Trim() "Roslyn\bincore\runtimes\win\native"
```

Comment:
What's this doing?

Reply:
diasymreader.dll does not come with .net core 3.0 and above SDKs. Since SDK base path was pointing to 3.x in devops, it was not picking up the dll thus failing crossgen. This logic picks up the latest 2.x available and gets the path to diasymreader.dll from there

---
In reply to: [393116917](https://github.com/Azure/azure-functions-host/pull/5781#discussion_r393116917) [](ancestors = 393116917)